### ISSUE TITLE
Queue modals call work not as expected

### ISSUE THREAD

**Issue by Mero-Plaform** [skeletonlabs/skeleton#2006]
### Current Behavior

`Not sure` if this is correct behavior or not, but looks weird and unusable for me.

I'm trying to call a new modal without closing the previous one.

Like if we have modal(let's call it modal1) to add new data and user pushes a button with an action that needs confirmation, in this case, I'm trying to open confirm modal(let's call it moal2) and return to the previous one(modal1) (ofc. I'm not waiting for data to be stored, it's on my job).

Right now such move will open same modal (modal1 will trigger modal1 to open again):

Open modal1 -> click inside to open modal2 -> modal1 has been opened again.

Looks like wrong modal been queued, because if I repeat attempt for opening modal2 from modal2 (let's say 5 times) after what I trigger closing current modal(click on backdrop or ESC) I see current modal to be closed and as previous modal is opened modal2 again and again (for those 5 times).

So it's like opening the wrong one but queueing the correct one.

### Expected Behavior

Parent modal should open the child modal and stay queued for reopening on the child modal close or parent modal should be closed on the child modal has been opened, this behavior is much clearer.

### Steps To Reproduce

The following code has been written in the project created by `pnpm create skeleton-app@latest my-skeleton-app`, with Bare Bones template, Skeleton theme & typescript, all other options were rejected.

By using it code you will see at top left screen corner button "open modal 1".
By clicking on "open modal 1" you will trigger modal 1 opening (type: component), in which would be text "modal 1" and button "open modal 2" which in turn will trigger modal 2 to open, but instead, you will see again modal1.
After clicking on "open modal 2" and closing the current modal (weather backdrop click or ESC button) you see modal2 to open. 
As many times you try to open modal1, after it closes the same number of times you see modal2 has opened.

+page.svelte file:
```
<script lang="ts">
  import { getModalStore, type ModalSettings } from "@skeletonlabs/skeleton";

  const modalStore = getModalStore();

  const openModal = () => {
    const modal: ModalSettings = {
      type: "component",
      component: "modal2",
    };
    modalStore.trigger(modal);
  };
</script>

<div>modal 1</div>

<button on:click={openModal}> open modal 2 </button>
```

Modal1.svelte file:

```
<script lang="ts">
  import { getModalStore, type ModalSettings } from "@skeletonlabs/skeleton";

  const modalStore = getModalStore();

  const openModal = () => {
    const modal: ModalSettings = {
      type: "component",
      component: "modal2",
    };
    modalStore.trigger(modal);
  };
</script>

<div>modal 1</div>

<button class="btn bg-slate-500" on:click={openModal}> open modal 2 </button>
```

Modal2.svelte file:
```
<div>modal 2</div>
```

### More Information

I'm getting this issue in `@skeletonlabs/skeleton: "1.10.0" & "2.0.0"`

**Comment by Sarenor** [comment]
Modals are a First in First Out list currently, not a stack. It's working as intended but we are considering transitioning to a stack.

**Comment by endigo9740** [comment]
To add a bit more context, this can be demonstrated on the doc site via the "queue multiple" example:
https://www.skeleton.dev/utilities/modals#trigger

If you would like to voice your support for the stacking approach, the open request for this is here:
https://github.com/skeletonlabs/skeleton/issues/1034

**Comment by Mero-Plaform** [comment]
Perhaps, but in your example here: "https://www.skeleton.dev/utilities/modals"

"Queue Multiple" button is opening new modal correctly and so it's closing the prew one automatically:

From page source link: 

```
function modalMultiple(): void {
		[1, 2, 3].forEach((dNum: number) => {
			const modal: ModalSettings = {
				type: 'alert',
				title: `Modal ${dNum}`,
				body: `The modal body of ${dNum}.`
			};
			modalStore.trigger(modal);
		});
	}
```

```
<button class="btn variant-filled" on:click={modalMultiple}>Queue Multiple</button>
````

In my case I'm getting opening again and again the first modal, which couldn't be correct as I see it

**Comment by endigo9740** [comment]
@Mero-Plaform looking at your code, you're seeing the same modal displayed twice because the same modal is triggered twice:

- `+page.svelte` -> triggers `component: "modal2"`
- `Modal1.svelte` -> triggers `component: "modal2"`

So there's no place in the examples provided where `Modal1` is triggered that's visible here.

**Comment by Mero-Plaform** [comment]
@endigo9740 you are right, something went wrong with code copy-paste, I've updated it

- `+page.svelte` -> triggers `component: "modal1"`
- `Modal1.svelte` -> triggers `component: "modal2"`

**Comment by endigo9740** [comment]
Gotcha. As an additional tip - I might suggest you find a location on your page to place:

```html
<pre class="pre">{JSON.stringify(modalStore, null, 2)}<pre>
```
> NOTE: you may need to use `$modalStore`, I'm away from my workstation and cannot confirm right now.

This will help you visualize the current queue, which can be useful in debugging and help understand how the queue works. But in essence it's a array like so:

```js
[
    modalInstance,
    modalInstance,
    modalInstance
]
```

Only the TOP most instance will be displayed at any given time. The `trigger()` method appends new modal instances to the bottom of the list. When you close a modal, it's pruned from the top of the array (from from index `0`) and the next in the list shows.

**Comment by Mero-Plaform** [comment]
@endigo9740 I get you. 

Your words about `trigger()` and `modal order` clarified the thing.

I was confused because in your example which I mentioned here: https://github.com/skeletonlabs/skeleton/issues/2006#issuecomment-1708922155 

Your modals seem to close before a new one appears in that example.

Is there any difference in triggering standard modals like `type: 'alert'` and `type: 'component'`?

**Comment by endigo9740** [comment]
> Your modals seem to close before a new one appears in that example.

Right. The modals are triggered in order: 1, 2, 3. So 1 is visible right away with 2/3 queued behind. When 1 closes 2 shows, when 2 closes 3 is shown. When 3 is closed the queue is empty and no modal is shown. It all happens to quick it may be hard to parse until you grasp what's going on.

> Is there any difference in triggering standard modals like type: 'alert' and type: 'component'?

Alert/confirm/prompt are just "built-ins" we provide to replace the native `window.alert()`, `window.confirm()`, and `window.prompt()`. But they are just treated as another modal instance as far as the queue is concerned. They can be mixed intermittently with `component` type modals.

If you're really curious about how this is handled, I might suggest a peak over the source code for the Modal component:
https://github.com/skeletonlabs/skeleton/blob/dev/packages/skeleton/src/lib/utilities/Modal/Modal.svelte#L239

In a nutshell we look at the "shape" of the data provided per instance, then construct or display the UI accordingly.

**Comment by mmagee-jbmetrics** [comment]
For others looking for a solution, instead of calling `modalStore.trigger()`, you can instead call the update function to place the new modal at the top of the queue in front of the current modal as below:
```
let modal: ModalSettings = { type: 'confirm', ...};
modalStore.update((mStore) => {
     mStore.unshift(modal);
     return mStore;
}
```
